src/dhali/transaction_utils.py

```diff
@@ -566,11 +569,66 @@ def move_document_in_transaction(transaction, source_ref, target_ref):
             transaction.delete(source_ref)
         else:
             return
+    except Exception as e:
+        logging.info(f"An error occured. Transaction reverted: {e}")
+        raise e
+
+def move_document(db, source_ref, destination_ref):
+    transaction = db.transaction()
+    try:
+        _move_document_in_transaction(transaction, source_ref, destination_ref)
     except KeyError as e:
-        # Assume that source_ref has been removed or never existed in the first place
-        logging.info(f'Source document does not exist: {e}')
+        logging.info(f'Expected KeyError: {e}')
```

Comment:
Maybe we want to be more explicit, e.g.: `document has already been moved, skipping...`

Reply:
Done